content/flux/v0.x/data-types/regexp.md

```diff
@@ -3,11 +3,139 @@ title: Work with regular expression types
+| :--- | :------------------------------------------------------------------------------ |
+| i    | case-insensitive                                                                |
+| m    | multi-line mode: `^` and `$` match begin/end line in addition to begin/end text |
+| s    | let `.` match `\n`                                                              |
+| U    | ungreedy: swap meaning of `x*` and `x*?`, `x+` and `x+?`, etc                   |
+
+Include regular expression flags at the beginning of your regular expression pattern
+enclosed in parentheses (`()`) and preceded by a question mark (`?`).
+
+```js
+/(?iU)foo*/
+```
+
+## Use regular expressions in predicate expressions
+To use regular expressions in [predicate expressions](/flux/v0.x/get-started/query-basics/#predicate-expressions),
+use the [`=~` and `!~` comparison operators](/flux/v0.x/spec/operators/#comparison-operators).
+The right operand must be a string.
+The left operand must be a regular expression.
+
+```js
+"abc" =~ /\w/
+// Returns true
+
+"z09se89" =~ /^[a-z0-9]{7}$/
+// Returns true
+
+"foo" !~ /^f/
+// Returns false
+
+"FOO" =~ /(?i)foo/
+// Returns true
+```
+
+## Convert a string to a regular expression
+1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp/).
+2. Use [`regexp.compile()`](/flux/v0.x/stdlib/regexp/compile) to compile a
+    string into a regular expression type.
+
+```js
+import "regexp"
+
+regexp.compile(v: "^- [a-z0-9]{7}")
+// Returns ^- [a-z0-9]{7} (regexp type)
+```
+
+## Examples
+
+- [Replace all substrings that match a regular expression](#replace-all-substrings-that-match-a-regular-expression)
+- [Return the first regular expression match in a string](#return-the-first-regular-expression-match-in-a-string)
+- [Escape regular expression metacharacters in a string](#escape-regular-expression-metacharacters-in-a-string)
+
+### Replace all substrings that match a regular expression
+1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
+2. Use [`regexp.replaceAllString()`](/flux/v0.x/stdlib/regexp/replaceallstring/)
+   and provide the following parameters:
+
+    - **r**: regular expression
+    - **v**: string to search
+    - **t**: replacement for matches to **r**.
+
+```js
+import "regexp"
+
+regexp.replaceAllString(
+  r: /a(x*)b/,
+  v: "-ab-axxb-",
+  t: "T"
+)
+// Returns "-T-T-"
+```
+
+### Return the first regular expression match in a string
+1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
+2. Use [`regexp.findString()`](/flux/v0.x/stdlib/regexp/findstring/) to return
+   the first regular expression match in a string.
+   Provide the following parameters:
+
+    - **r**: regular expression
+    - **v**: string to search  
+
+```js
+import "regexp"
+
+regexp.findString(
+  r: /foo.?/,
+  v: "seafood fool"
+)
+// Returns "food"
+```
+
+### Escape regular expression metacharacters in a string
+1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
```

Comment:
Maybe add a bit for context here?

```suggestion
Certain characters must be escaped in order to use them as literals in regular expressions.
This can make the expression hard to read.
Flux provides a convenience method `regex.quoteMeta()`.

1. Import the [`regexp` package](/flux/v0.x/stdlib/regexp).
```